### PR TITLE
fix: minor fixes in Gratuity

### PR DESCRIPTION
**Pull request by Anurag810**
Checkbox for payment method.
removed Validation for Payroll date and Relieving Date.

Exception handling:
Screenshot 2021-05-21 at 2 46 27 PM

 
Form Clean up
Screenshot 2021-05-21 at 3 06 57 PM

### CODE REVIEW

File: erpnext/payroll/doctype/gratuity/gratuity.json [hunk 1]
```diff
@@ -162,41 +156,48 @@
...
-   "mandatory_depends_on": "eval: doc.pay_via_salary_slip == 0",
+   "mandatory_depends_on": "eval: doc.payment_method == \"Payment Entry\"",
    "options": "Cost Center"
+  },
+  {
+   "default": "Additional Salary",
+   "fieldname": "payment_method",
+   "fieldtype": "Select",
+   "label": "Payment Method",
+   "options": "Additional Salary\nPayment Entry"
```
Comment (ruchamahabal) [inline erpnext/payroll/doctype/gratuity/gratuity.json:194]: We will need a patch to update the payment method in existing documents?

Comment (Anurag810) [reply erpnext/payroll/doctype/gratuity/gratuity.json:194]: done

File: erpnext/patches/v13_0/set_gratuity_payment_method.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from __future__ import unicode_literals
+import frappe
+
+def execute():
+    frappe.reload_doctype("Gratuity")
+
+    frappe.db.sql("update `tabGratuity` set `payment_method` = 'Additional Salary' where `pay_via_salary_slip` = 1")
+    frappe.db.sql("update `tabGratuity` set `payment_method` = 'Payment Entry' where `pay_via_salary_slip` = 1")
```
Comment (ruchamahabal) [inline erpnext/patches/v13_0/set_gratuity_payment_method.py]:
```suggestion
    frappe.db.sql("update `tabGratuity` set `payment_method` = 'Additional Salary' where `pay_via_salary_slip` = 1")
    frappe.db.sql("update `tabGratuity` set `payment_method` = 'Payment Entry' where `pay_via_salary_slip` = 0")
```

Comment (Anurag810) [reply erpnext/patches/v13_0/set_gratuity_payment_method.py]: @ruchamahabal oops.

**Comment by Anurag810** [comment]
@ruchamahabal patch is not related to this pr

File: erpnext/patches/v13_0/set_gratuity_payment_method.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+from __future__ import unicode_literals
+import frappe
+
+def execute():
+    frappe.reload_doctype("Gratuity")
```
Comment (ruchamahabal) [inline erpnext/patches/v13_0/set_gratuity_payment_method.py]:
```suggestion
    frappe.reload_doc("payroll", "doctype", "gratuity")
```

Comment (Anurag810) [reply erpnext/patches/v13_0/set_gratuity_payment_method.py]: @ruchamahabal I think both are the same..

**Comment by Anurag810** [comment]
@ruchamahabal Closing this Pull Request because it needs to be redesigned for Full and final settlement Feature.

**Pull request closed without merging**